drivers/hmis/app/graphql/types/hmis_schema/enums/project_config_type.rb

```diff
@@ -0,0 +1,23 @@
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+module Types
+  class HmisSchema::Enums::ProjectConfigType < Types::BaseEnum
+    graphql_name 'HmisProjectConfigType'
+
+    def self.get_project_config_type(cls_name)
```

Comment:
One thing I'd like feedback on -- do you think this code is being a bit too clever for its own good and perhaps sacrificing readability in pursuit of DRY-ness?

Reply:
Oh, I see how the value is working now too, with the `input.config_type.constantize` in the mutation. That feels a bit overly clever to me too tbh. I think just for readability, my preference would be to change `CreateProjectConfig` to just use a `case` statement on the expected values for `input.config_type` (AUTO_EXIT, AUTO_ENTER), and raise an error if its an unrecognized type. I think it will make it easier for future-selves-and-others to understand how project configs work.

```
  class_name = case input.config_type
      when 'AUTO_ENTER"
        Hmis::ProjectAutoEnterConfig
      when 'AUTO_EXIT"
        Hmis::AutoExitConfig
      else
        raise "Unsupported type: #{input.config_type}"
      end
```